src/bot-command-handlers.js

```diff
@@ -41,36 +28,179 @@ module.exports = {
-            }
-
-            rolls[specs[i]] = internalRolls;
+function createRollMessages(args){
+
+    let messages = [];
+
+    let separateDieRolls = []
+
+    // create a default specification roll if nothing sent to us
+
+    if (args.message.trim().length === 0) {
+        let spec = rollSpecification('');
+        let rollResults = rollDice(spec['numDie'], spec['dieSize']);
+        let message = rollResultMessage(spec, rollResults);
+
+        messages.push(message);
+    }
+    else {
+        separateDieRolls = args.message.split(/[ ,]+/); // split by comma or whitespace
+    }
+
+    // loop through all the separate dice rolls
+    for (let i = 0; i < separateDieRolls.length; i++) {
+    
+        let spec = rollSpecification(separateDieRolls[i]);
+        let rollResults = rollDice(spec['numDie'], spec['dieSize']);
+        let message = rollResultMessage(spec, rollResults); 
+
+        messages.push(message);
+    }
+
+    return messages;
+}
+
+function rollResultMessage(spec, rolls){
+  
+    let message = spec['desc'] + ': ';
+    // create the result of the roll on one line
+    // outcomes can be total, best, worst with possibility of modifier  
+    // spec['desc'] is the roll we are creating 
+
+    // display messages:
+    // w2d10+1: 5, 7 -> worst: 5  modified: 6
+    // 2d6: 2 + 5 = 7
+    // 2d10-1: 5 + 8 = 13 modified: 12
+    // b2d6: 2, 5 -> best: 5
+
+
+    let total = rolls.reduce(function(tot, num) {
+        return tot + num;
+    });
+
+    let best = Math.max.apply(null, rolls);
+    let worst = Math.min.apply(null, rolls);
+    let baseValue = 0;
+
+    if (spec['prefix'] === 'w' || spec['prefix'] === 'b') {
+        message = message + rolls.join(', ') + ' -> ';
+        if (spec['prefix'] === 'b'){
+            message = message + 'best: ' + best;
+            baseValue = best;
+        }
+        else {
+            message = message + 'worst: ' + worst;
+            baseValue = worst;
+        }
+    }
+    else {  // total is the default
+        message = message + rolls.join (' + ') + ' = ' + total;
+        baseValue = total;
+    }
+   
+    // create modified value if required
+
+    if (spec['modAmount'] > 0) {
+        let modifier = Number(spec['modifier'] + spec['modAmount']);
+        let modified = 0;
+        modified = baseValue + modifier;
+
+        if (modified < 0 ){
+            modified = 0;
         }
+
+        message = message + ' modified: ' + modified;
     }
 
+    return message;
+
+}
+
+
+function rollSpecification(roll) {
+
+    //create a specification of the dice roll
+
+    //a default die roll spec 1d6 with no modifier and no worst/best option
+    
+    let prefix = null;
+    let numDie = 1;
+    let dieSize = 6;
+    let modifier = null;
+    let modAmount = 0;
+
+    let spec = {};
+    spec['prefix'] = prefix;
+    spec['numDie'] = numDie;
+    spec['dieSize'] = dieSize;
+    spec['modifier'] = modifier;
+    spec['modAmount'] = modAmount;  
+    spec['desc'] = "";
```

Comment:
X-D Are you trying to get paid by line? Why not just...
```
let spec = {
    "prefix" = null,
    "numDie" = 1,
    "dieSize" = 6,
    "modifier" = null,
    "modAmount" = null,
    "desc" = ""
};
```

That would cause below to need updating to...
```
    if (rollInfo.length > 1 ) {
        spec['prefix'] = (typeof rollInfo[1] != 'undefined') ? rollInfo[1].toLowerCase() : spec['prefix'];
        spec['numDie'] = (typeof rollInfo[2] != 'undefined') ? rollInfo[2] : spec['numDie'];
        spec['dieSize'] = (typeof rollInfo[3] != 'undefined') ? rollInfo[3] : spec['dieSize'];
        spec['modifier'] = (typeof rollInfo[4] != 'undefined') ? rollInfo[4] : spec['modifier'];
        spec['modAmount'] = (typeof rollInfo[5] != 'undefined') ? rollInfo[5] : spec['modAmount'];
    }
```

Or another option which I feel is a little bit easier to read / maintain is...
```
    let normalize = function(spec, key, data, shouldLowCase){
        if (typeof data != 'undefined'){
            spec[key] = shouldLowCase ? data.toLowerCase() : data;
        }
    };
   
    if (rollInfo.length > 1 ) {
        normalize(spec, 'prefix', rollInfo[1], true);
        normalize(spec, 'numDie', rollInfo[2], false);
        normalize(spec, 'dieSize', rollInfo[3], false);
        normalize(spec, 'modifier', rollInfo[4], false);
        normalize(spec, 'modAmount', rollInfo[5], false);
    }
```